lib/gtfs_realtime_viz/state.ex

```diff
@@ -31,22 +29,31 @@ defmodule GTFSRealtimeViz.State do
     GenServer.call(pid, {:vehicles, group, vehicles, comment})
   end
 
-  @spec vehicles(GenServer.server, term) :: [Proto.vehicle_position]
+  @spec vehicles(GenServer.server, term) :: [{String.t, [Proto.vehicle_position]}]
   def vehicles(pid \\ __MODULE__, group)
   def vehicles(pid, group) do
-    GenServer.call(pid, {:vehicles, group})
+    pid
+    |> GenServer.call({:vehicles, group})
+    |> Enum.reverse
   end
 
   # server callbacks
 
   def handle_call({:vehicles, group, vehicles, comment}, _from, state) do
     new_state = update_in(state, [Access.key(group, [])], fn prev_msgs ->
-      Enum.take([{comment, vehicles} | prev_msgs], @max_archive)
+      msgs = [{comment, vehicles} | prev_msgs]
+      if max_archive() == :infinity do
+        msgs
+      else
+        Enum.take(msgs, max_archive())
```

Comment:
Small thing, but you could set the value of `max_archive()` to a variable so the calculation only needs to be made at most once, even though it's a pretty low stakes operation.

Reply:
Sure, good idea!